src/test/emulators/storage/files.spec.ts

```diff
@@ -157,6 +196,229 @@ describe("files", () => {
+            decodedObjectId: "dir%2Fobject",
+            originalUrl: "localhost:9000",
+            ttlSeconds: 1.4,
+          })
+        ).to.be.rejectedWith(BadRequestError);
+      });
+
+      it("should throw an error if TTL is below the min or above the max time", () => {
+        const storageLayer = getStorageLayer(ALWAYS_TRUE_RULES_VALIDATOR);
+
+        const cases = [SIGNED_URL_MIN_TTL_SECONDS, SIGNED_URL_MAX_TTL_SECONDS];
+
+        cases.forEach((time) => {
+          expect(
+            storageLayer.generateSignedUrl({
+              bucketId: "10",
+              decodedObjectId: "dir%2Fobject",
+              originalUrl: "localhost:9000",
+              ttlSeconds: time === SIGNED_URL_MAX_TTL_SECONDS ? SIGNED_URL_MAX_TTL_SECONDS + 1 : 0,
+            })
+          ).to.be.rejectedWith(BadRequestError);
+        });
+      });
+
+      it("should throw an error if request is not authorized", () => {
+        const storageLayer = getStorageLayer(ALWAYS_FALSE_RULES_VALIDATOR);
+
+        expect(
+          storageLayer.generateSignedUrl({
+            bucketId: "10",
+            decodedObjectId: "dir%2Fobject",
+            originalUrl: "localhost:9000",
+            ttlSeconds: 10,
+          })
+        ).to.be.rejectedWith(ForbiddenError);
+      });
+
+      it("should throw an error if object does not exist", () => {
+        const storageLayer = getStorageLayer(ALWAYS_TRUE_RULES_VALIDATOR);
+        expect(
+          storageLayer.generateSignedUrl({
+            bucketId: "10",
+            decodedObjectId: "dir%2Fobject",
+            originalUrl: "localhost:9000",
+            ttlSeconds: 10,
+          })
+        ).to.be.rejectedWith(NotFoundError);
+      });
+
+      it("should return a valid signed URL", async () => {
+        const storageLayer = getStorageLayer(ALWAYS_TRUE_RULES_VALIDATOR);
+        const regexToReplace = /X-Firebase-Date=([^&]+)/;
+
+        await uploadFile(storageLayer, "bucket", "dir/object", {
+          data: "Hello, World!",
+          metadata: { contentType: "mime/type" },
+        });
+
+        const tempUnsignedUrl = createUnsignedUrl({
+          bucketId: "bucket",
+          decodedObjectId: "dir%2Fobject",
+          urlTtlSeconds: 1,
+          urlUsableSeconds: "*",
+          url: "localhost:9000",
+        });
+
+        const signedUrlObject = await storageLayer.generateSignedUrl({
+          bucketId: "bucket",
+          decodedObjectId: "dir%2Fobject",
+          originalUrl: "localhost:9000",
+          ttlSeconds: 1,
+        });
+
+        const signedDate = regexToReplace.exec(signedUrlObject.signed_url);
+
+        const unsignedUrl = `${tempUnsignedUrl.replace("X-Firebase-Date=*", signedDate![0])}`;
+
+        const signature = createSignature(unsignedUrl);
+
+        const actualSignedUrl = `${unsignedUrl}&X-Firebase-Signature=${encodeURIComponent(
+          signature
+        )}`;
+
+        expect(signedUrlObject.signed_url).to.equal(actualSignedUrl);
```

Comment:
In this case `signedUrlObject` is the "actual" value, `actualSignedUrl` is the expectation right?

Reply:
I was thinking of it as I am testing signedUrlObject, so in my method I expect it to be the actualSignedUrl that I know is correct

Reply:
In typical testing language, "actual*" is the thing that gets produced by the logic under test, and "expected*" is the hard-coded expectation.

Reply:
in the same file, there was a test that has 
` expect(metadata.contentType).to.equal("mime/type");
        expect(data.toString()).to.equal("Hello, World!");`
so I just followed the format that was already there. Should I change the tests in the file as well? I feel like it would be confusing if we test both ways